cpp/include/cuvs/neighbors/ivf_pq.hpp

```diff
@@ -470,17 +498,22 @@ struct index : cuvs::neighbors::index {
   raft::device_vector_view<const uint32_t, uint32_t, raft::row_major> list_sizes() const noexcept;
 
   /** Cluster centers corresponding to the lists in the original space [n_lists, dim_ext] */
-  raft::device_matrix_view<float, uint32_t, raft::row_major> centers() noexcept;
   raft::device_matrix_view<const float, uint32_t, raft::row_major> centers() const noexcept;
+  /** Setter for the centers [n_lists, dim_ext] or [n_lists, dim] */
+  void update_centers(raft::resources const& handle,
+                      raft::device_matrix_view<const float, uint32_t, raft::row_major> new_centers);
 
   raft::device_matrix_view<const int8_t, uint32_t, raft::row_major> centers_int8(
     const raft::resources& res) const;
   raft::device_matrix_view<const half, uint32_t, raft::row_major> centers_half(
     const raft::resources& res) const;
 
   /** Cluster centers corresponding to the lists in the rotated space [n_lists, rot_dim] */
-  raft::device_matrix_view<float, uint32_t, raft::row_major> centers_rot() noexcept;
   raft::device_matrix_view<const float, uint32_t, raft::row_major> centers_rot() const noexcept;
+  /** Setter for the cluster centers in the rotated space [n_lists, rot_dim] */
+  void update_centers_rot(
```

Comment:
Can we have these update functions in the `cuvs::neighbors::ivf_pq::helpers` namespace? (consistent with `set_centers`)

Reply:
I think it would introduce a lot of redonduncy for nothing. These update functions are public and accessible so users might as well directly call them.